### DISCUSSION TITLE
Continuous Backup: Deletion & Truncation Directory

### DISCUSSION THREAD

**Discussion by MutenRoshi84** [microstream-one/microstream#264] [Q&A] [answered]
Hello,

I set up the continuous backup as in https://docs.microstream.one/manual/storage/configuration/backup/continuous-backup.html.

**Now I have the following questions:**

truncation directory:
- What is the truncation directory for and is it absolutely necessary? I configured it, but MicroStream has never created it for me.

deletion directory:
- Is the deletion directory automatically emptied or do I have to do it myself?
- If the deletion directory is automatically emptied, what does it depend on and where / how can I configure it?
- If I have to empty the deletion directory myself, can I do this while the storage is running? Do I have to pay attention to something?

**Comment by hg-ms** [comment, marked as answer]
Hello,

If those directories are configured Microstream will store additional backup data there.
Enabling them is not required. In most use cases the normal “Backup” is enough. Especially since enabling the deletion directory might increase the required storage-space dramatically.
Also it is not trivial restoring data backuped in those directories as this must be done manually… 

Deletion directory:
If configured Microstream will not delete storage files. Instead of deleting they are copied into the deletion-directory (e.g. if the housekeeping refactors files containing to much deleted data).
The deletion directory is not emptied automatically.
It is legal to delete it’s content. Microstream won’t access those files anymore.

Truncation directory:
If Microstream cuts of a part of a storage file (e.g. after an incomplete write gets rolled back) an additional backup of that trucated data is kept in the truncation dir.

**Reply by MutenRoshi84** [reply]
Hello,

Thank you for the insightful feedback!

But I still have one question.
I have restored a backup from the "Backup" folder as a test. I deleted the original files of the DB and moved the backed up files from the "Backup" folder to the original directory. Is that the right approach or do I have to pay attention to something else?

**Comment by hg-ms** [comment]
Your approach is fine.
The Continuous Backup as well as the Full Backup are copies of the original storage. 

You may also copy Full Backup’s files to another location if you like to archive it.
With the continuous backup you need to shut down the store before copying as some files may be locked.
